Examples/Framework/include/ActsExamples/EventData/Measurement.hpp

```diff
@@ -282,19 +289,56 @@ class MeasurementProxyBase {
     return self().subspaceHelper().expandMatrix(self().covariance());
   }
 
+  /// @brief Construct the measurement from a sourcelink, subspace vector,
+  /// parameters, and covariance.
+  ///
+  template <typename SubspaceDerived, typename ParameterDerived,
+            typename CovarianceDerived>
+  void construct(const Acts::SourceLink& source_link,
+                 const Eigen::DenseBase<SubspaceDerived>& subspace,
+                 const Eigen::DenseBase<ParameterDerived>& parameters,
+                 const Eigen::DenseBase<CovarianceDerived>& covariance)
+    requires(!ReadOnly) && requires {
+      { this->self().subspaceIndexVector() = subspace };
+      { this->self().parameters() = parameters };
+      { this->self().covariance() = covariance };
+    }
+  {
+    setSourceLink(source_link);
+    self().subspaceIndexVector() = subspace;
+    self().parameters() = parameters;
+    self().covariance() = covariance;
+  }
+
+  /// @brief Construct the measurement from a sourcelink, subspace vector,
+  /// parameters, and covariance.
+  ///
+  template <typename OtherDerived>
+  void construct(const OtherDerived& other)
+    requires(!ReadOnly) && requires {
+      { other.size() } -> std::integral;
+      { other.sourceLink() };
+      { other.subspaceIndexVector() };
+      { other.parameters() };
+      { other.covariance() };
+    }
```

Comment:
1. should we create a named concept for this?
2. also what kind of object can be fed into this?

Reply:
Basically just any measurement proxy.